provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/Location.java

```diff
@@ -1,14 +1,99 @@
 package gov.va.api.health.providerdirectory.api.resources;
 
-public class Location {
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import gov.va.api.health.providerdirectory.api.Fhir;
+import gov.va.api.health.providerdirectory.api.bundle.AbstractBundle;
+import gov.va.api.health.providerdirectory.api.bundle.AbstractEntry;
+import gov.va.api.health.providerdirectory.api.bundle.BundleLink;
+import gov.va.api.health.providerdirectory.api.datatypes.Address;
+import gov.va.api.health.providerdirectory.api.datatypes.CodeableConcept;
+import gov.va.api.health.providerdirectory.api.datatypes.Coding;
+import gov.va.api.health.providerdirectory.api.datatypes.ContactPoint;
+import gov.va.api.health.providerdirectory.api.datatypes.Identifier;
+import gov.va.api.health.providerdirectory.api.datatypes.Period;
+import gov.va.api.health.providerdirectory.api.datatypes.Signature;
+import gov.va.api.health.providerdirectory.api.datatypes.SimpleResource;
+import gov.va.api.health.providerdirectory.api.elements.BackboneElement;
+import gov.va.api.health.providerdirectory.api.elements.Element;
+import gov.va.api.health.providerdirectory.api.elements.Extension;
+import gov.va.api.health.providerdirectory.api.elements.Meta;
+import gov.va.api.health.providerdirectory.api.elements.Narrative;
+import gov.va.api.health.providerdirectory.api.elements.Reference;
+import io.swagger.v3.oas.annotations.media.Schema;
+import java.util.List;
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
 
-    Status status;
+@Data
+@Builder
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+@Schema(
+        description = "http://www.fhir.org/guides/argonaut/pd/StructureDefinition-argo-location.html"
+)
+public class Location implements DomainResource {
+    @NotBlank String resourceType;
 
-    String name;
+    @Pattern(regexp = Fhir.ID)
+    String id;
+
+    @Valid Meta meta;
+
+    @Pattern(regexp = Fhir.URI)
+    String implicitRules;
+
+    @Pattern(regexp = Fhir.CODE)
+    String language;
+
+    @Valid Narrative text;
+
+    @Valid List<SimpleResource> contained;
+
+    @Valid List<Extension> extension;
+
+    @Valid List<Extension> modifierExtension;
+
+    @Valid List<Identifier> identifier;
+
+    @NotNull Status status;
+
+    Coding operationalStatus;
+
+    @NotNull String name;
+
+    List<String> alias;
 
     String description;
 
     Mode mode;
+
+    @Valid CodeableConcept type;
+
+    @Valid @NotNull List<ContactPoint> telecom;

```

Comment:
So spec says this list needs to be of size one or higher.

Maybe add comment that the transformer should check that at least one entry is in the list.

Just having the `telcom` not being null won't check that it has an entry in it.